n3fit/src/evolven3fit_new/evolve.py

```diff
@@ -0,0 +1,221 @@
+        dump_evolved_replica(
+            evolved_block, usr_path, int(replica.removeprefix("replica_"))
+        )
+    # remove folder:
+    # The function dump_evolved_replica dumps the replica files in a temporary folder
+    # We need then to remove it after fixing the position of those replica files
+    (usr_path / "nnfit" / usr_path.stem).rmdir()
+
+
+def load_fit(usr_path):
+    """
+    Loads all the replica pdfs at fitting scale in usr_path and return the exportgrids
+
+    Parameters
+    ----------
+
+        usr_path: pathlib.Path
+            path to the folder containing the fit
+
+    Returns
+    -------
+
+            : dict
+            exportgrids info
+    """
+    nnfitpath = usr_path / "nnfit"
+    pdf_dict = {}
+    for yaml_file in nnfitpath.glob("replica_*/*.exportgrid"):
+        data = yaml.safe_load(yaml_file.read_text(encoding="UTF-8"))
+        pdf_dict[yaml_file.parent.stem] = data
+    return pdf_dict
+
+
+def evolve_exportgrid(exportgrid, eko, x_grid):
+    """
+    Evolves the provided exportgrid for the desired replica with the eko and returns the evolved block
+
+    Parameters
+    ----------
+        exportgrid: dict
+            exportgrid of pdf at fitting scale
+        eko: eko object
+            eko operator for evolution
+        xgrid: list
+            xgrid to be used as the targetgrid
+    Returns
+    -------
+        : np.array
+        evolved block
+    """
+    # construct LhapdfLike object
+    pdf_grid = np.array(exportgrid["pdfgrid"]).transpose()
+    pdf_to_evolve = utils.LhapdfLike(pdf_grid, exportgrid["q20"], x_grid)
+    # evolve pdf
+    evolved_pdf = apply.apply_pdf(eko, pdf_to_evolve)
+    # generate block to dump
+    targetgrid = list(eko["targetgrid"])
+
+    def ev_pdf(pid, x, Q2):
+        return x * evolved_pdf[Q2]["pdfs"][pid][targetgrid.index(x)]
+
+    block = genpdf.generate_block(
+        ev_pdf,
+        xgrid=targetgrid,
+        Q2grid=list(eko["Q2grid"]),
+        pids=br.flavor_basis_pids,
+    )
+    return block
+
+
+def dump_evolved_replica(evolved_block, usr_path, replica_num):
+    """
+    Dump the evolved replica given by evolved_block as the replica num "replica_num" in
+    the folder usr_path/nnfit/replica_<replica_num>/usr_path.stem.dat
+
+    Parameters
+    ----------
+        evolved_block: numpy.array
+            block of an evolved PDF
+        usr_path: pathlib.Path
+            path of the fit folder
+        replica_num: int
+            replica number
+    """
+    path_where_dump = usr_path / "nnfit" / usr_path.stem
+    # create folder to dump the evolved replica if it does not exist
+    path_where_dump.mkdir(exist_ok=True)
+    to_write_in_head = f"PdfType: replica\nFromMCReplica: {replica_num}\n"
+    genpdf.export.dump_blocks(
+        path_where_dump, replica_num, [evolved_block], pdf_type=to_write_in_head
+    )
+    # fixing_replica_path
+    utils.fix_replica_path(usr_path, replica_num)
```

Comment:
I am a bit confused here: Why are we writing to some path and then immediately after moving it to another one?

Reply:
As already discussed, we are using some utility functions coming from `genpdf` to dump the replicas and the info file. These functions assume that you want to write stuff in a certain folder that is not what we want in this case. Still, since we know that they are working and since we want to avoid duplicating functions that we already have, I believe it is better to rely on them and then fixing the paths.

Reply:
That seems a bit unsatisfactory... Why are we walking around a piece of code whose main purpose should be to enable this? Can we not modify it instead?

Reply:
The main purpose of `genpdf` is not to support `evolven3fit`. It has been developed to manipulate and generate LHAPDF sets, mostly fake ones, for theory development.
Incidentally, generating LHAPDF sets is also useful in the context of postfit, thus the same functions are going to be reimplemented.

The utility there could be generalized, but as a first shot, adapting the already available result is not that bad.

Reply:
I have the impression this is duplicating things we already have. Seems to me this looks a lot like valifdphys.lhio.write_replica. I'd much rather have one set of tools than two, where we have to work around one of them.